snapcraft/plugins/go.py

```diff
@@ -177,6 +178,16 @@ def build(self):
         for package in packages:
             binary = os.path.join(self._gopath_bin, self._binary_name(package))
             self._run(["go", "build", "-o", binary] + tags + [package])
+            # Relink with system linker if executable is dynamic in order to be
+            # able to set rpath later on. This workaround can be removed after
+            # https://github.com/NixOS/patchelf/issues/146 is fixed.
+            if self._is_classic:
+                if elf.ElfFile(path=binary).is_dynamic:
```

Comment:
is there a reason for double nesting instead of using `and` here?

Reply:
No special reason, will turn into a single conditional.